pkg/auth/cell.go

```diff
@@ -26,27 +26,54 @@ var Cell = cell.Module(
 	cell.ProvidePrivate(
 		// Null auth handler provides support for auth type "null" - which always succeeds.
 		newNullAuthHandler,
+		// CT map authenticator provides support to write authentication information into the eBPF conntrack map
+		newCtMapAuthenticator,
 	),
 )
 
 type authManagerParams struct {
 	cell.In
 
-	EndpointManager endpointmanager.EndpointManager
-	AuthHandlers    []authHandler `group:"authHandlers"`
+	AuthHandlers          []authHandler `group:"authHandlers"`
+	DatapathAuthenticator datapathAuthenticator
 }
 
 type Manager interface {
 	consumer.MonitorConsumer
+	SetIPCache(*ipcache.IPCache)
 }
 
 func newManager(params authManagerParams) (Manager, error) {
-	manager, err := newAuthManager(params.EndpointManager, params.AuthHandlers)
+	mgr, err := newAuthManager(params.AuthHandlers, params.DatapathAuthenticator, nil)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create auth manager: %w", err)
 	}
 
-	return monitor.New(manager), nil
+	return &manager{
+		monitorConsumer: monitor.New(mgr),
+		manager:         mgr,
```

Comment:
`manager.manager` feels confusing to me, might want to name those more specific

Reply:
as mentioned: the goal is to have the `IPCache` injected via hive once this is possible (should be soon). then the explicit setter `SetIPCache` & this whole struct can be deleted :pray: